business_requirement/models/business_requirement.py

```diff
@@ -0,0 +1,260 @@
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', False)]}
+    )
+    confirmation_date = fields.Datetime(
+        copy=False,
+        readonly=True
+    )
+    confirmed_id = fields.Many2one(
+        'res.users', string='Confirmed by',
+        copy=False,
+        readonly=True
+    )
+    reviewed_date = fields.Datetime(
+        copy=False,
+        readonly=True
+    )
+    reviewed_id = fields.Many2one(
+        'res.users',
+        string='Reviewed by',
+        copy=False,
+        readonly=True
+    )
+    approval_date = fields.Datetime(
+        copy=False,
+        readonly=True
+    )
+    approved_id = fields.Many2one(
+        'res.users',
+        string='Approved by',
+        copy=False,
+        readonly=True
+    )
+    company_id = fields.Many2one(
+        'res.company', string='Company',
+        required=True, readonly=True, states={'draft': [('readonly', False)]},
+        default=lambda self: self.env.user.company_id,
+    )
+
+    @api.multi
+    @api.onchange('project_id')
+    def project_id_change(self):
+        for br in self:
+            if br.project_id and br.project_id.partner_id:
+                br.partner_id = br.project_id.partner_id
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', '/') == '/':
+            vals['name'] = self.env['ir.sequence'].\
+                next_by_code('business.requirement')
+        return super(BusinessRequirement, self).create(vals)
+
+    @api.multi
+    @api.depends('parent_id')
+    def _get_level(self):
+        for br in self:
+            level = br.parent_id and br.parent_id.level + 1 or 1
+            br.level = level
+
+    @api.multi
+    @api.depends('business_requirement_ids')
+    def _compute_sub_br_count(self):
+        for br in self:
+            br.sub_br_count = len(br.business_requirement_ids)
+
+    @api.model
+    def _get_states(self):
+        states = [
+            ('draft', 'Draft'),
+            ('confirmed', 'Confirmed'),
+            ('approved', 'Approved'),
+            ('stakeholder_approval', 'Stakeholder Approval'),
+            ('in_progress', 'In progress'),
+            ('done', 'Done'),
+            ('cancel', 'Cancel'),
+            ('drop', 'Drop'),
+        ]
+        return states
+
+    @api.multi
+    def action_button_confirm(self):
+        self.state = 'confirmed'
+        self.confirmed_id = self.env.user
+        self.confirmation_date = fields.Datetime.now()
+
+    @api.multi
+    def action_button_back_draft(self):
+        self.state = 'draft'
+        self.confirmed_id = self.approved_id = []
+        self.confirmation_date = self.approval_date = ''
+
+    @api.multi
+    def action_button_approve(self):
+        self.state = 'approved'
+        self.approved_id = self.env.user
+        self.approval_date = fields.Datetime.now()
+
+    @api.multi
+    def action_button_stakeholder_approval(self):
+        self.state = 'stakeholder_approval'
```

Comment:
Modification not required anymore, changing the name https://github.com/sudhir-serpentcs/business-requirement/pull/9 is enough